test/prop_thoas.erl

```diff
@@ -0,0 +1,236 @@
+       begin
+           Encoded = thoas:encode(Input),
+           ?assertEqual(Input, jsx:decode(Encoded)),
+           true
+       end).
+
+prop_jsx_encode_thoas_decode_match() ->
+    ?FORALL(
+       Input,
+       json_term(),
+       begin
+           Encoded = jsx:encode(Input),
+           ?assertEqual({ok, Input}, thoas:decode(Encoded)),
+           true
+       end).
+
+%% Not realy reliable
+%% prop_encode_not_slower_than_jsx() ->
+%%     ?FORALL(
+%%        Input,
+%%        ?SIZED(Size, json_term(Size + 10)),
+%%        begin
+%%            erlang:garbage_collect(),
+%%            {JsxTime, _} = timer:tc(run_n_f(fun() -> jsx:encode(Input) end, 50)),
+%%            erlang:garbage_collect(),
+%%            {ThoasTime, _} = timer:tc(run_n_f(fun() -> thoas:encode(Input) end, 50)),
+%%            ?assert(ThoasTime / JsxTime < 2,
+%%                    [{thoas_time, ThoasTime},
+%%                     {jsx_time, JsxTime}]),
+%%            true
+%%        end).
+
+%% prop_decode_not_slower_than_jsx() ->
+%%     ?FORALL(
+%%        Input,
+%%        ?SIZED(Size, json_term(Size + 10)),
+%%        begin
+%%            JsonBin = thoas:encode(Input),
+%%            erlang:garbage_collect(),
+%%            {JsxTime, _} = timer:tc(run_n_f(fun() -> jsx:decode(JsonBin) end, 50)),
+%%            erlang:garbage_collect(),
+%%            {ThoasTime, _} = timer:tc(run_n_f(fun() -> thoas:decode(JsonBin) end, 50)),
+%%            ?assert(ThoasTime / JsxTime < 2,
+%%                    [{thoas_time, ThoasTime},
+%%                     {jsx_time, JsxTime}]),
+%%            true
+%%        end).
+
+%% run_n_f(Fun, N) ->
+%%     fun() -> run_n(Fun, N) end.
+
+%% run_n(_Fun, 0) -> ok;
+%% run_n(Fun, N) -> 
+%%     Fun(),
+%%     run_n(Fun, N - 1).
+
+%% Generator
+
+encode_options() ->
+    oneof(
+      [
+       #{},
+       ?LET(Escape,
+            oneof([json, unicode, html, javascript]),
+            #{escape => Escape})
+      ]).
+
+decode_options() ->
+    oneof(
+      [
+       #{},
+       ?LET(Strings,
+            oneof([reference, copy]),
+            #{strings => Strings})
+      ]).
+-define(DIV_FACTOR, 4).
+
+%% flexible representation (accepted by encoder, but can be altered by decoder)
+input_term() ->
+  ?SIZED(Size, input_term(Size)).
+
+input_term(Size) ->
+  ?LAZY(proper_types:frequency(
+          [{40, json_term(Size)},
+           {10, ?SUCHTHAT(Atom, atom(), not lists:member(Atom, [true, false, null]))},
+           {20, j_array(Size, fun input_term/1)},
+           {10, j_object(Size, fun input_term/1)},
+           {20, proplist_object(Size)}
+          ])).
+
+proplist_object(0) ->
+    #{};                                        %[] would be confused for array
+proplist_object(Size) ->
+    proper_types:list({proper_types:oneof([j_string(), atom()]),
+                       input_term(Size div ?DIV_FACTOR)}).
+
+%% Canonical representation (returned by decoder)
+json_term() ->
+  ?SIZED(Size, json_term(Size)).
```

Comment:
```suggestion
    ?SIZED(Size, json_term(Size)).
```

Reply:
btw, do you use some code formatter? If so, maybe I can just run it over this module as well…

Reply:
I do not but I would be open to using one. I'm not sure what one is best these days.

Reply:
I'm using WhatsApp's `erlfmt` for personal projects. For now just fixed the indentation to be consistent - don't want to pivot the PR too much from just introducing the tests